example/main.go

```diff
@@ -37,21 +39,51 @@ func main() {
 		},
 	}
 
-	s := mcp.NewServer(serverInfo, tools)
+	prompts := []mcp.PromptDefinition{
+		{
+			Metadata: mcp.Prompt{
+				Name:        "example",
+				Description: ptr("An example prompt template"),
+				Arguments: []mcp.PromptArgument{
+					{
+						Name:        "text",
+						Description: ptr("Text to process"),
+						Required:    ptr(true),
+					},
+				},
+			},
+			Process:   processPrompt,
+			RateLimit: rate.NewLimiter(rate.Every(time.Second), 5),
+		},
+	}
+
+	s := mcp.NewServer(serverInfo, tools, prompts)
 	s.Serve()
 }
 
-func computeSHA256(params mcp.CallToolRequestParams) (mcp.CallToolResult, error) {
+func ptr[T any](t T) *T {
+	return &t
+}
+
+// Update the computeSHA256 function to send a single notification
```

Comment:
This comment looks like a TODO item. It could be removed or changed to document the function.